homeassistant/components/api/__init__.py

```diff
@@ -369,19 +374,27 @@ async def post(self, request, domain, service):
             )
 
         context = self.context(request)
+        changed_states: list[ReadOnlyDict[str, Collection[Any]]] = []
+
+        @ha.callback
+        def _async_save_changed_entities(
+            event: EventType[EventStateChangedData],
+        ) -> None:
+            if event.context == context and (state := event.data["new_state"]):
+                changed_states.append(state.as_dict())
```

Comment:
Why do we change from storing the states to storing the state data dicts?

Reply:
It's going to call as_dict anyways when the json serializer is run. By converting it ahead of time we avoid the overhead of the json default fallback to figure out how to serialize the data since it's already a dict